### DISCUSSION TITLE
关于 use ff::PrimeField; 和  arithmetic::Field,  用哪个？

### DISCUSSION THREAD

**Discussion by Demian101** [z2o-k7e/halo2-step-by-step#51] [Q&A] [answered]
看到课程代码里有 2 种版本的 Field:

```rust
use halo2_proofs::arithmetic::Field

// VS :

use ff::PrimeField; 
```

虽然都不报错，但是比较疑惑：
1. 使用这个有什么讲究吗？ 什么情况下需要用哪种形式？
2. 哪个更好呢？

**Comment by pochenai** [comment, marked as answer]
PrimeField有F::From( v as u64), F::ONE，这样的API; Field没有

**Comment by Demian101** [comment]
哦哦 ，然后像 

`use halo2_proofs::{arithmetic::FieldExt,` 这种东西是不是已经过时了，被扔进了时代的垃圾桶

**Reply by pochenai** [reply]
嗯，反正最新的库里都没有这个了，Halo2的有一个更新里好像有提到这个，你可以找下
